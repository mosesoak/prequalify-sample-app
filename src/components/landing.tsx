import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { submitQualifyForm } from '../state/qualifyForm/actions';
import {
  qualifyFormChangeActionType,
  QualifyFormState,
} from '../state/qualifyForm/types';
import { AppState } from '../state/rootReducer';

export const Landing = () => {
  const dispatch = useDispatch();
  let {
    loading,
    error,
    price,
    make,
    model,
    annualIncome,
    creditScore,
  } = useSelector((state: AppState) => state.qualifyFormReducer);

  useEffect(() => {
    if (error) {
      alert(error);
    }
  });

  const submitForm = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault(); // otherwise React warns that form is not connected
    dispatch(submitQualifyForm());
  };

  if (loading) {
    return <div>One moment...</div>;
  }

  const handleChange = (event: React.FormEvent<HTMLInputElement>) => {
    const { id, value } = event.currentTarget;
    dispatch(
      qualifyFormChangeActionType({
        id: id as keyof QualifyFormState,
        value: value,
      }),
    );
  };

  return (
    <form id="qualify-form" onSubmit={submitForm}>
      <label>
        Auto Price
        <span className="input-prefix">$</span>
        <input
          required
          id="price"
          type="text"
          onChange={handleChange}
          value={price?.toLocaleString('en') || ''}
        />
      </label>

      <label>
        Auto Make
        <span className="input-prefix" />
        <input
          required
          id="make"
          type="text"
          onChange={handleChange}
          value={make || ''}
        />
      </label>

      <label>
        Auto Model
        <span className="input-prefix" />
        <input
          required
          id="model"
          type="text"
          onChange={handleChange}
          value={model || ''}
        />
      </label>

      <label>
        Your Estimated Annual Income
        <span className="input-prefix">$</span>
        <input
          required
          id="annualIncome"
          type="text"
          onChange={handleChange}
          value={annualIncome || ''}
        />
      </label>

      <label>
        Your Credit Score
        <span className="input-prefix" />
        <input
          required
          id="creditScore"
          type="text"
          onChange={handleChange}
          value={creditScore?.toLocaleString('en') || ''}
        />
      </label>

      <div>
        <input type="submit" value="Submit Form" />
      </div>
    </form>
  );
};
