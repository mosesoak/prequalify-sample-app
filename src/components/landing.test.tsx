import React from 'react';
import renderer from 'react-test-renderer';
import { Landing } from './landing';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

// Shows a simple snapshot testing setup for components with a mock redux store

const middlewares: any[] = [];
const mockStore = configureStore(middlewares);
const initialState = {
  systemReducer: {
    isPrequalified: false,
    notQualifiedReason: undefined,
    isLoggedIn: false,
  },
  qualifyFormReducer: {
    loading: false,
    price: 123,
    make: 'foo',
    model: 'bar',
    annualIncome: 123,
    creditScore: 123,
  },
};

test('Page renders correctly', () => {
  const store = mockStore(initialState);
  const tree = renderer
    .create(
      <Provider store={store}>
        <Landing />
      </Provider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
