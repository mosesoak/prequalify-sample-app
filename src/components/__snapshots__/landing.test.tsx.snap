// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders correctly 1`] = `
<form
  id="qualify-form"
  onSubmit={[Function]}
>
  <label>
    Auto Price
    <span
      className="input-prefix"
    >
      $
    </span>
    <input
      id="price"
      onChange={[Function]}
      required={true}
      type="text"
      value="123"
    />
  </label>
  <label>
    Auto Make
    <span
      className="input-prefix"
    />
    <input
      id="make"
      onChange={[Function]}
      required={true}
      type="text"
      value="foo"
    />
  </label>
  <label>
    Auto Model
    <span
      className="input-prefix"
    />
    <input
      id="model"
      onChange={[Function]}
      required={true}
      type="text"
      value="bar"
    />
  </label>
  <label>
    Your Estimated Annual Income
    <span
      className="input-prefix"
    >
      $
    </span>
    <input
      id="annualIncome"
      onChange={[Function]}
      required={true}
      type="text"
      value={123}
    />
  </label>
  <label>
    Your Credit Score
    <span
      className="input-prefix"
    />
    <input
      id="creditScore"
      onChange={[Function]}
      required={true}
      type="text"
      value="123"
    />
  </label>
  <div>
    <input
      type="submit"
      value="Submit Form"
    />
  </div>
</form>
`;
